remote/remote_test.go

```diff
@@ -1491,7 +1496,7 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 				h.AssertNil(t, err)
 
 				h.AssertEq(t, img.Found(), false)
-				h.AssertError(t, img.Delete(), "MANIFEST_UNKNOWN")
+				h.AssertError(t, img.Delete(), "NAME_UNKNOWN")
```

Comment:
These are explicitly different errors [here](https://github.com/google/go-containerregistry/blob/83f4080544c8578ec02b937c2a0a4dc07198916d/pkg/registry/manifest.go#L96-L111). The `NAME_UNKNOWN` is if the entire repo doesn't exist, whereas `MANIFEST_UNKNOWN` means a specific manifest under that repo is missing.

It seems weird the entire repo is gone after we delete...

Reply:
Ah, interesting. I assumed this was just something different about `ggcr/registry`'s cleanup mechanisms but perhaps it's a bug in `Delete` that `docker-distribution` was covering up. I'll see if I can set up some test scenarios to figure out what's going on.

Reply:
This ends up being consistent with `gcr.io` behavior. Using a ref like,`gcr.io/dotnet-build-gcp/notarepo:latest`, where `dotnet-build-gcp` is a project but `notarepo:latest` doesn't exist and doing something like:
```go
image, err := remote.NewImage("gcr.io/dotnet-build-gcp/notarepo:latest", authn.DefaultKeychain)
image.Delete() // returns `NAME_UNKNOWN`
```

Output:
```
2021/05/07 07:25:29 DELETE https://gcr.io/v2/dotnet-build-gcp/neverthere/manifests/sha256:03cbce912ef1a8a658f73c660ab9c539d67188622f00b15c4f15b89b884f0e10: NAME_UNKNOWN: Failed to compute blob liveness for manifest: 'sha256:03cbce912ef1a8a658f73c660ab9c539d67188622f00b15c4f15b89b884f0e10'
exit status 1
```